babel_edtf/__init__.py

```diff
@@ -174,6 +175,23 @@ def _format_edtf0_interval_naive(edtf_interval, format, locale):
         dt_start, dt_end, skeleton, fuzzy=True, locale=locale)
 
 
+def _validate_leap_year(edtf_date):
+    """Validate that dates on the 29th of February are on leap years."""
+    if isinstance(edtf_date, Date):
+        if (
+            edtf_date.precision == PRECISION_DAY
+            and edtf_date.day == '29'
+            and edtf_date.month == '02'
+            and not calendar.isleap(int(edtf_date.year))
```

Comment:
Remark: About the `int(edtf_date.year)` bit, I found [one place in python-edtf where they do a more complex conversion to support imprecise years](https://github.com/ixc/python-edtf/blob/master/edtf/parser/parser_classes.py#L264-L268); not sure if it's worth it to handle this case?